src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -249,3 +259,47 @@ def snapshot(self) -> list[ClusterInfo]:
     def _try_fetch(self) -> list[ClusterInfo]:
         for row in self._fetch(f"SELECT * FROM {self._schema}.{self._table}"):
             yield JobInfo(*row)
+
+
+class JobsRunCrawler(CrawlerBase):
+    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
+        super().__init__(sbe, "hive_metastore", schema, "job_runs")
+        self._ws = ws
+
+    def _crawl(self) -> list[JobRunInfo]:
+        no_of_days_back = datetime.timedelta(days=30)  # todo make configurable in yaml?
+        start_time_from = datetime.datetime.now() - no_of_days_back
+        # todo figure out if we need to specify a default timezone
+        all_job_runs = list(
+            self._ws.jobs.list_runs(start_time_from=start_time_from, start_time_to=datetime.datetime.now())
+        )
+        all_clusters: dict[str, ClusterDetails] = {c.cluster_id: c for c in self._ws.clusters.list()}
+        return self._assess_job_runs(all_clusters, all_job_runs)
+
+    def _assess_job_runs(self, all_clusters, all_job_runs):
+        all_job_run_info = []
+        job_runs_without_job_id = list(filter(lambda jr: jr.job_id is None, all_job_runs))
+        for job_run in job_runs_without_job_id:
+            for job_run_cluster in job_run.job_clusters:
+                cluster_key = job_run_cluster.job_cluster_key
+                cluster: ClusterDetails = (
+                    all_clusters[cluster_key] if cluster_key is not None else job_run_cluster.new_cluster
```

Comment:
a) the logic is wrong - `cluster_key` is an alias, that is valid in scope of a job, not an id. `existing_cluster_id ` is something that has real cluster id.
b) there's a logic for getting cluster confs - for example, in the job assessment. Why don't you reuse that? :) 

<img width="688" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/790b5bba-099e-4928-b250-d786026b4108">

https://docs.databricks.com/api/workspace/jobs/create

Reply:
I believe this has been changed and is even more complete, but I am not using the code in the job assessment because it's in an internal method, do you have a pattern for reusing code you want us to follow? Util class etc.?